content/ui-patterns/notification-messaging.mdx

```diff
@@ -0,0 +1,174 @@
+- A non-destructive action with side-effects like costs or access changes is about to take place
+
+<img
+  width="960"
+  alt="A global warning banner above GitHub's global navigation bar inside a browser window."
+  src="https://github.com/primer/design/assets/18661030/468b73b1-de8b-45e0-b8bb-325516df942e"
+/>
+
+### Success
+
+Use the `success` state to notify users that the results of an action were successful if it is not apparent from other parts of the UI. **Use success messaging sparingly** and rely more on interaction context.
+
+Examples:
+- Saving an account setting if there's no page reload or redirect
+- A background process has successfully completed
+
+### Critical
+
+Use the `critical` state to notify users that an action was unsuccessful, to display form errors, or to inform that an action is destructive.
+
+Examples:
+- Form field is invalid upon submitting
+- A delete or transfer button
+
+<img
+  width="960"
+  alt="A critical error message on an input field within GitHub Issues."
+  src="https://github.com/primer/design/assets/18661030/5da2e479-6bc7-46dc-9179-46002412b9f5"
+/>
+
+### Neutral
+
+Reserve the `neutral` state for **inline** system related degraded experiences, or for surfacing account related permission messages.
+
+Examples:
+- A row of a data table could not be loaded
+- An item in a list could not be loaded
+- User does not have permissions to perform an action
+
+See [degraded experiences](./degraded-experiences) for more information.
+
+### Upsell
+
+Use the `upsell` state to highlight features that could be unlocked if the user upgrades their account.
+
+Examples:
+- Upgrade account to access a feature
+
+<img
+  width="960"
+  alt="An upsell banner inside the rulesets setting page on GitHub explaing that this feature can be used if the user upgrades to an Enterprise account."
+  src="https://github.com/primer/design/assets/18661030/924884b8-4b85-4967-898d-e0c2fcfaf785"
+/>
+
+## Placement
+
+Message proximity contributes to the prominence of a message. Whenever possible, place messages near the action they are related to.
+
+### Above global navigation
+
+Announcements displayed above the global navigation are reserved for site wide issues that require immediate attention. These messages are not dismissable and should only be used for critical issues that impact the entire site.
+
+<img
+  width="960"
+  alt="Highlights a global announcement banner above the GitHub global navigation bar inside a browser window."
+  src="https://github.com/primer/design/assets/18661030/3d843939-5100-424e-a9a4-d663c517432f"
+/>
+
+### Top of body
+
+If messaging pertains to an entire page or section, place it at the top of the body content. This is the most common placement for messaging components. Banners should fill the container width but maintain appropriate spacing between elements and not appear full width or flush under the global nav.
```

Comment:
In case there's a sidebar/sidenav on the page, should the container span the width of the **main content** or the entire page?